katana/katana.py

```diff
@@ -91,6 +93,208 @@ def readconfig():
     return cfg
 
 
+@route('/searchkw', method='POST')
+def searchkw():
+    """
+    This method returns the ActionFile Path of the selected Driver.
+    """
+    value = parseString("".join(request.body))
+    tree = get_correct_xml_and_root_element(value)
+    driver_name = tree.text
+    driver_file = gpysrcdir + '/ProductDrivers/' + driver_name + ".py"
+    actiondir_new = mkactiondirs(driver_file)
+    py_files = mkactionpyfiles(actiondir_new)
+    return py_files
+
+
+def get_correct_xml_and_root_element(value):
+    """ To get the correct xml format from Katana UI and to get it's root element """
+    indented_xml = "".join(value.toprettyxml(newl='\n'))
+    corrected_xml = remove_extra_newlines_char_xml(indented_xml)
+    tree = xml.etree.ElementTree.fromstring(corrected_xml)
+    return tree
+
+
+@route('/parsexmlobj', method='POST')
+def parsexmlobj():
+    """
```

Comment:
Break down this function into smaller units. Create separate methods to perform repeated operations(ex.  line 202 to 215 and line 223 to 234 looks similar). Use more meaning variable names ex. kw_name instead of kw_list(line 158).

Reply:
Done, please check.

Reply:
Changed the logic and removed the repeated operations. Please check.